storage/database/db_manager.go

```diff
@@ -1744,59 +1749,33 @@ func (dbm *databaseManager) DeleteCode(hash common.Hash) {
 	}
 }
 
-// Cached Trie Node operation.
-func (dbm *databaseManager) ReadCachedTrieNode(hash common.Hash) ([]byte, error) {
+func (dbm *databaseManager) ReadTrieNode(hash common.Hash) ([]byte, error) {
 	dbm.lockInMigration.RLock()
 	defer dbm.lockInMigration.RUnlock()
 
 	if dbm.inMigration {
-		if val, err := dbm.GetStateTrieMigrationDB().Get(hash[:]); err == nil {
+		if val, err := dbm.ReadTrieNodeFromNew(hash); err == nil {
 			return val, nil
 		} else if err != dataNotFoundErr {
 			// TODO-Klaytn-Database Need to be properly handled
 			logger.Error("Unexpected error while reading cached trie node from state migration database", "err", err)
 		}
 	}
-	val, err := dbm.ReadCachedTrieNodeFromOld(hash)
+	val, err := dbm.ReadTrieNodeFromOld(hash)
 	if err != nil && err != dataNotFoundErr {
 		// TODO-Klaytn-Database Need to be properly handled
 		logger.Error("Unexpected error while reading cached trie node", "err", err)
 	}
 	return val, err
 }
 
-// Cached Trie Node Preimage operation.
-func (dbm *databaseManager) ReadCachedTrieNodePreimage(secureKey []byte) ([]byte, error) {
-	dbm.lockInMigration.RLock()
-	defer dbm.lockInMigration.RUnlock()
-
-	if dbm.inMigration {
-		if val, err := dbm.GetStateTrieMigrationDB().Get(secureKey); err == nil {
-			return val, nil
-		}
-	}
-	return dbm.ReadCachedTrieNodePreimageFromOld(secureKey)
-}
-
-// State Trie Related operations.
-func (dbm *databaseManager) ReadStateTrieNode(key []byte) ([]byte, error) {
-	dbm.lockInMigration.RLock()
-	defer dbm.lockInMigration.RUnlock()
-
-	if dbm.inMigration {
-		if val, err := dbm.GetStateTrieMigrationDB().Get(key); err == nil {
-			return val, nil
-		}
-	}
-	return dbm.ReadStateTrieNodeFromOld(key)
-}
-
-func (dbm *databaseManager) HasStateTrieNode(key []byte) (bool, error) {
-	val, err := dbm.ReadStateTrieNode(key)
+func (dbm *databaseManager) HasTrieNode(hash common.Hash) (bool, error) {
```

Comment:
Wouldn't it be more appropriate to return only bool?
If we need the reason (error) too, we can call `ReadTrieNode` instead.

Reply:
I agree HasTrieNode need not return an error.
But I decided to limit the impact of this PR as in [this comment](https://github.com/klaytn/klaytn/pull/1846#issuecomment-1576007705).

Already existing functions (though slightly changed) keep the original interfaces (i.e. ReadTrieNode and HasTrieNode)
and newly introduced functions use the error-less interface (i.e. WriteTrieNode, DeleteTrieNode, PutTrieNodeToBatch).

Reply:
OK, thanks.